docs/sources/reference/commandline/cli.md

```diff
@@ -533,6 +533,10 @@ considered as normal files. Filepaths in .dockerignore are absolute with
 the current directory as the root. Wildcards are allowed but the search
 is not recursive.
 
+When the `URL` references a Dockerfile, or a Dockerfile is read from 
+`STDIN`, then the `-f` option will have no effect and the contents of the
+Dockerfile will be placed in a file called `Dockerfile`.
+
```

Comment:
@duglin Text between bars is a rewrite of that cli.md and is intended for that file. I think it solves your problem rewrite and clarifies some other structure problems in that section.  If you don't want to add it to the command that's fine. If you do, let me know, happy to email you the markup.

Regarding the API, no, I did not want to copy the entire section on the Git repository.  I do think the mention of "git" in the command form without any explanation isn't useful as it lacks context   I'll tweak that line comment.